pkg/kubelet/kuberuntime/logs/logs_test.go

```diff
@@ -454,3 +453,67 @@ func TestReadLogsLimitsWithTimestamps(t *testing.T) {
 
 	assert.Equal(t, 2, lineCount, "should have two lines")
 }
+
+func TestReadLogsWithLogRotation(t *testing.T) {
+	logLineFmt := "2023-06-09T05:28:57.806368292Z stdout F %v\n"
+
+	dir, err := os.MkdirTemp("", "readlogs")
+	assert.NoError(t, err)
+	defer os.RemoveAll(dir)
+	tmpfile, err := os.CreateTemp(dir, "log.*.txt")
```

Comment:
```suggestion
	tmpfile, err := os.CreateTemp(t.TempDir(), "log.*.txt")
```

Reply:
It seems that `fsnotify` may lose events if there are insufficient permissions for `t.TempDir()`, at least on my Mac OS. Creating a new directory is ok.

Reply:
That sounds interesting, considering that `t.TempDir` [creates a directory with `0777` permissions](https://cs.opensource.google/go/go/+/refs/tags/go1.20.5:src/testing/testing.go;drc=38cfb3be9d486833456276777155980d1ec0823e;l=1223). Can you give more details?

Reply:
Sorry, I mistakenly interpreted `os.CreateTemp(t.TempDir(), "log.*.txt")` as `os.CreateTemp("", "log.*.txt")`.
It's not fine with `os.CreateTemp("", "log.*.txt")`.

Fixed.